qir/qat/Passes/StaticResourceComponent/AllocationAnalysisPass.cpp

```diff
@@ -29,7 +29,7 @@ bool AllocationAnalysisPass::extractResourceId(llvm::Value* value, uint64_t& ret
             return false;
         }
 
-        llvm::Type* element_type = pointer_type->getElementType();
```

Comment:
Do we still need lines 26 to 30 above? There's no other use of `pointer_type` with this change. Or do you still need the null check on line 27 (i.e. you can't just check if element_type is null here)

Reply:
@billti FYI, we eventually have to completely change this code for opaque pointers.